src/util/filesystem.cxx

```diff
@@ -0,0 +1,134 @@
+#include <filesystem.hxx>
+#include <Settings.hxx>
+#include <LOG.hxx>
+#include <fstream>
+
+#include <SDL.h>
+
+namespace fs
+{
+
+std::string readFileAsString(const std::string &fileName, bool binaryMode)
+{
+#ifdef __ANDROID__
+  // on Android, files cannot be accessed directly without Java. SDL implements that for us, so we use SDL RWOps until we implemented JNI File Access ourselves
+  // binary mode is ignored for now, savegames can't be loaded
+
```

Comment:
We probably want to prepend fs::getBasePath to fileName, and get rid of the desktop implementation below assuming it will be a portable implementation for android and desktop

Reply:
i don't understand what you mean.

On android, getBasePath now returns an empty string. I verified that a basePath is not needed on Android, so think this should be ok?